internal/cli/iam/projects/create.go

```diff
@@ -74,51 +76,90 @@ func (opts *CreateOpts) newRegionUsageRestrictions() string {
 }
 
 func (opts *CreateOpts) newCreateProjectOptions() *atlas.CreateProjectOptions {
-	return &atlas.CreateProjectOptions{ProjectOwnerID: opts.projectOwnerID}
+	return &atlas.CreateProjectOptions{
+		ProjectOwnerID: opts.projectOwnerID}
 }
 
 func (opts *CreateOpts) validateOwnerID() error {
 	if opts.projectOwnerID == "" {
 		return nil
 	}
 
-	if config.Service() != config.OpsManagerService {
+	sv, constrain, err := opts.newOpsManagerConstrain()
+	if err != nil {
+		return err
+	}
+
+	if sv == nil || constrain == nil {
 		return nil
 	}
 
-	v, err := opts.store.ServiceVersion()
-	if err != nil {
-		return err
+	if !constrain.Check(sv) {
+		return fmt.Errorf("%s is available only for Atlas, Cloud Manager and Ops Manager >= 6.0", flag.OwnerID)
 	}
 
-	sv, err := cli.ParseServiceVersion(v)
+	return nil
+}
+
+func (opts *CreateOpts) validateWithoutDefaultAlertSettings() error {
+	if !opts.withoutDefaultAlertSettings {
+		return nil
+	}
+
+	sv, constrain, err := opts.newOpsManagerConstrain()
 	if err != nil {
 		return err
 	}
 
-	constrain, _ := semver.NewConstraint(">= 6.0")
+	if sv == nil || constrain == nil {
+		return nil
+	}
 
 	if !constrain.Check(sv) {
-		return fmt.Errorf("%s is available only for Atlas, Cloud Manager and Ops Manager >= 6.0", flag.OwnerID)
+		return fmt.Errorf("%s is available only for Atlas, Cloud Manager and Ops Manager >= 6.0", flag.WithoutDefaultAlertSettings)
 	}
 
 	return nil
 }
 
-// mongocli iam project(s) create <name> [--orgId orgId] [--ownerID ownerID].
+func (opts *CreateOpts) newOpsManagerConstrain() (*semver.Version, *semver.Constraints, error) {
```

Comment:
[nit] I can understand wanting to reduce duplication but this feels like the wrong abstraction, main reason this is the bad approach `opts.store.ServiceVersion()` is an API call, you are calling that twice when it could be reused and those making this command slower (and more error prone introducing more network calls), main point here the service version should be reused across feature validations

Reply:
updated